### PR TITLE
[libtasn1]  Update to version 4.18. Fixes JB#57060

### PR DESCRIPTION
**Pull request by Thaodan**

### CODE REVIEW

**Comment by vigejolla** [comment]
The submodule still points to gitlab.com instead of our mirror

File: rpm/libtasn1.spec [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Name:       libtasn1
 Summary:    This is the ASN.1 library used in GNUTLS
-Version:    4.13
+Version:    4.18.0
 Release:    1
 Group:      System/Libraries
```
Comment (saukko) [inline rpm/libtasn1.spec]: Extra points cutting out groups.

File: rpm/libtasn1.spec [hunk 2]
```diff
@@ -30,7 +36,6 @@ This package contains tools using the libtasn library.
 
 %package devel
 Summary:    Files for development of applications which will use libtasn1
-Group:      Development/Libraries
 Requires:   %{name} = %{version}-%{release}
 Requires(post): /sbin/install-info
```
Comment (Tomin1) [inline rpm/libtasn1.spec]: These install-info dependencies in devel package are a leftover from the past. They should be removed, I think.

File: rpm/libtasn1.spec [hunk 3]
```diff
@@ -56,19 +60,34 @@ Man and info pages for %{name}.
 %setup -q -n %{name}-%{version}/%{name}
```
Comment (Tomin1) [inline rpm/libtasn1.spec]: You probably should change this to use %autosetup. Now the patches are not applied because there are no %patch macros.

**Review by Tomin1 — CHANGES_REQUESTED** [review]
Some changes needed.

**Comment by Thaodan** [comment]
Adjusted

File: rpm/libtasn1.spec [hunk 4]
```diff
@@ -44,31 +49,43 @@ use libtasn1.
 
 %package doc
 Summary:   Documentation for %{name}
-Group:     Documentation
 Requires:  %{name} = %{version}-%{release}
-Requires(post): /sbin/install-info
```
Comment (Tomin1) [inline rpm/libtasn1.spec:49]: You removed these from doc and not from devel, like I said.

**Review by Tomin1 — CHANGES_REQUESTED** [review]

**Comment by Tomin1** [comment]
This seems to require libtool (`libtoolize`) to build. It didn't build after `sfdk build-requires reset`.

Also `%check` failed at "`FAIL: reproducers`" (after I had installed libtool to target snapshot).

**Comment by Thaodan** [comment]
> This seems to require libtool (`libtoolize`) to build. It didn't build after `sfdk build-requires reset`.

Fixed.

> Also %check failed at "FAIL: reproducers" (after I had installed libtool to target snapshot).

That's because the package version isn't matching without tag (check log).

**Comment by Tomin1** [comment]
> > This seems to require libtool (`libtoolize`) to build. It didn't build after `sfdk build-requires reset`.
> 
> Fixed.
> 
> > Also %check failed at "FAIL: reproducers" (after I had installed libtool to target snapshot).
> 
> That's because the package version isn't matching without tag (check log).

Ah, okay. So, what I actually needed is `-c no-fix-version`. :+1:

**Review by Tomin1 — APPROVED** [review]

**Comment by martyone** [comment]
> > > Also %check failed at "FAIL: reproducers" (after I had installed libtool to target snapshot).
>>
> > That's because the package version isn't matching without tag (check log).
>
> Ah, okay. So, what I actually needed is `-c no-fix-version`. +1

From a brief check I would expect that this is addressed by https://github.com/sailfishos/libtasn1/blob/e439b365467a923441b190bb1fa3f0bfafb5d1b2/rpm/libtasn1.spec#L65 - maybe something more could be done here.

**Comment by Tomin1** [comment]
> > > > Also %check failed at "FAIL: reproducers" (after I had installed libtool to target snapshot).
> > > 
> > > 
> > > That's because the package version isn't matching without tag (check log).
> > 
> > 
> > Ah, okay. So, what I actually needed is `-c no-fix-version`. +1
> 
> From a brief check I would expect that this is addressed by
> 
> https://github.com/sailfishos/libtasn1/blob/e439b365467a923441b190bb1fa3f0bfafb5d1b2/rpm/libtasn1.spec#L65
> - maybe something more could be done here.

As I understand it, the issue was that when I didn't have a tag, it used `4.13+git2+something` as the version so I ended up with `4.13` in `.tarball-version` which didn't pass the test. When I used `no-fix-version`, then I get the version in spec file and that passes the test. I should get the same result by tagging the commit I'm building, right?

**Comment by martyone** [comment]
You are right. Having to tag everything you want to build would be very inconvenient. Use of `no-fix-version` is a valid workaround but still not optimal. The preferred approach would be to ensure that packages build well with the default options.

Anyway, I checked more and in this particular case the test is failing just because the package was updated to a "much newer" version and the failure will not happen during regular changes after this particular PR is merged and tagged, so LGTM.

**Comment by Thaodan** [comment]
> > > > Also %check failed at "FAIL: reproducers" (after I had installed libtool to target snapshot).
> > > 
> > > 
> > > That's because the package version isn't matching without tag (check log).
> > 
> > 
> > Ah, okay. So, what I actually needed is `-c no-fix-version`. +1
> 
> From a brief check I would expect that this is addressed by
> 
> https://github.com/sailfishos/libtasn1/blob/e439b365467a923441b190bb1fa3f0bfafb5d1b2/rpm/libtasn1.spec#L65
> - maybe something more could be done here.

This to avoid making the package/gnulib using the inbuild mechanism  to detect the version based on git.

If mb2 resets the version based on the last tags its to bold for the check as noticed by @martyone  above.

**Pull request merged**